### PR TITLE
watchOS 10: fix crash

### PR DESCRIPTION
**Pull request by leandroalonso**
Since we started using Xcode 16, the app crashes every time for users on watchOS 10.6.1.

After a lot (A LOT, REALLY A LOT) of tries, I pinpointed the culprit to the `[weak self]`. I'm still very puzzled as to why this happens in first place.

I changed the code to remove the `[weak self]` for watchOS 10 and only watchOS 10.

## To test

1. Code review
2. I'll test this using a TestFlight build as Xcode builds don't present the issue

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: Modules/Server/Sources/PocketCastsServer/Public/Refresh/RefreshManager.swift [hunk 1]
```diff
@@ -78,6 +82,40 @@ public class RefreshManager {
             }
         }
     }
+    #endif
+
+    #if os(watchOS)
```
Comment (leandroalonso) [inline Modules/Server/Sources/PocketCastsServer/Public/Refresh/RefreshManager.swift:87]: Not proud of this code duplication (triplication?). But I have no better idea :(

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/Automattic/pocket-casts-ios/milestone/74">7.78.2</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by danielebogo — APPROVED** [review]
🤞 🤞 🤞

File: Modules/Server/Sources/PocketCastsServer/Public/Refresh/RefreshManager.swift [hunk 2]
```diff
@@ -78,6 +82,40 @@ public class RefreshManager {
...
+    // Watch OS 10 has a very very very weird issue that happened after using Xcode 16:
+    // if the refresh call uses a [weak self] the system kills the app.
+    // I'm not entirely sure why that happens but removing the weak part fixes the issue.
+    // We can safely remove this method once we remove support to watchOS 10.
```
Comment (DanReyLop) [inline Modules/Server/Sources/PocketCastsServer/Public/Refresh/RefreshManager.swift:91]: Please don't do that for a while. The iPhone app will happily update itself even if the associated watchOS app becomes incompatible, and I haven't found a reasonable way to rollback an app update (I've looked for it over the last 3 months).

Comment (leandroalonso) [reply Modules/Server/Sources/PocketCastsServer/Public/Refresh/RefreshManager.swift:91]: @DanReyLop we have no intention to remove support for watchOS 10, don't worry :)

**Pull request merged**